runtime/doc/lsp.txt

```diff
@@ -663,6 +676,119 @@ client_is_stopped({client_id})                   *vim.lsp.client_is_stopped()*
+                       filetypes = { 'lua' },
+                       cmd = { 'lua-language-server' },
+                       root_markers = { '.luarc.json', 'luarc.json' },
+                       single_file_support = true,
+                     },
+                     gopls = {
+                       filetypes = { 'go', 'gomod' },
+                       cmd = { 'gopls' },
+                       root_markers = { 'go.mod' },
+                       settings = {
+                         gopls = {
+                           analyses = {
+                             unusedparams = true,
+                           },
+                         },
+                       },
+                     },
+                   }
+                 })
+<
+
+                Parameters: ~
+                    {config}  (table|nil) Configuration table with two keys:
+                              "defaults" and "servers". The "servers" table
+                              takes a list of key-value pairs mapping
+                              (arbitrary) server names to server
+                              configurations. Both the "defaults" table as
+                              well as the individual server configuration
+                              tables accept any of the keys listed in
+                              |vim.lsp.start_client()|. The keys specified in
+                              "defaults" will act as default values for all
+                              servers unless overriden by a configuration in
+                              "servers". The "root_dir" option can also be a
+                              function, in which case it is passed the file
+                              path and buffer number of the buffer and should
+                              return the root directory to use. In addition to
+                              the keys listed in |vim.lsp.start_client()| the
+                              following keys are recognized:
+                              • filetypes (table): This option is only valid
+                                for server configurations in "servers". A list
+                                of filetypes for which this server
+                                configuration should map to.
+                              • autostart (boolean or function, default true):
+                                Automatically start or attach an LSP client
+                                for new buffers. When a function, it is called
+                                with the file name and buffer number of the
+                                buffer and should return a boolean indicating
+                                if the client should autostart. When autostart
+                                is false, a configured LSP server can be
+                                started manually using |vim.lsp.buf_start()|.
+                              • root_markers (table): A list of file or
```

Comment:
Here's a bikeshedding opportunity: right now the way this works is that `root_markers` overrides the values in the `defaults` table (the same as every other option). So if you want to use a common set of root markers (e.g. `.git`, `.hg`, etc.) you need to add those explicitly to each server configuration.

This is obviously tedious and annoying, so I'm open to suggestions. Possible solutions:

1. Make `root_markers` a special case that combines with the value in `defaults` (not a fan of this, because special cases are confusing and then there's no way to *disable* the default root markers)
2. Add a separate option like `common_root_markers` or `use_vcs_root` that is separate from the `root_markers` option. I think this is much better than (1), but still not great
3. Just live with it and tell people to do this:

   ```lua
   local vcs = { ".git", ".hg", ".svn"}
   vim.lsp.config({ servers = {
     clangd = {
       root_markers = { "compile_commands.json", "compile_flags.txt", unpack(vcs) },
     },
   }})
   ```